crates/astria-sequencer/src/service/mempool.rs

```diff
@@ -79,6 +98,18 @@ fn handle_check_tx(req: request::CheckTx) -> response::CheckTx {
     // TODO(https://github.com/astriaorg/astria/issues/228): status codes for various errors
     // TODO(https://github.com/astriaorg/astria/issues/319): offload `check_stateless` using `deliver_tx_bytes` mechanism
     //       and a worker task similar to penumbra
+    match transaction::check_nonce_mempool(&signed_tx, &state).await {
+        Ok(_) => (),
+        Err(e) => {
+            return response::CheckTx {
+                code: AbciCode::INVALID_PARAMETER.into(),
```

Comment:
Should this be an `INVALID_NONCE` introduced in #364 ?

Reply:
yes! thanks